src/hotspot/share/interpreter/interpreterRuntime.cpp

```diff
@@ -273,7 +273,14 @@ JRT_ENTRY(void, InterpreterRuntime::uninitialized_static_inline_type_field(JavaT
           Handle(THREAD, klass->protection_domain()),
           true, CHECK);
       assert(field_k != nullptr, "Should have been loaded or an exception thrown above");
-      klass->set_inline_type_field_klass(index, InlineKlass::cast(field_k));
+      if (!field_k->is_inline_klass()) {
+        THROW_MSG(vmSymbols::java_lang_IncompatibleClassChangeError(),
+                  err_msg("class %s expects class %s to be a concrete value class but it is not",
+                  klass->name()->as_C_string(), field_k->external_name()));
+      }
+      InlineLayoutInfo* li = klass->inline_layout_info_adr(index);
+      li->set_klass(InlineKlass::cast(field_k));
+      li->set_kind(LayoutKind::REFERENCE);
```

Comment:
Do you always set these together?  Might be good to have one function like fill_in() that fills both values in.

Reply:
Not always. For instance fields, the class of those fields is usually known while parsing the class file (LoadableDescriptors attribute, @NullRestricted annotation), but the layout kind is only known after the field layout computation (post-processing of the parsed stream).